### PR TITLE
#1163 - Check for square matrix in isinvertible

### LINKED ISSUE #1163: "Error: matrix is not square" in inv(M) in AbstractPolyhedron linear_map
The problem:
```julia
M = rand(5,4)
LazySets.isinvertible(M) #true
inv(M) #error
```

Should `linear_map` with `check_invertibility = true` use `pinv` instead of `inv` to deal with non square matrices?

### PR DESCRIPTION
**Pull request by schillic**
Closes #1163.

### CODE REVIEW

**Comment by mforets** [comment]
I suggest to change the default value of `use_inv`:

```julia
use_inv::Bool=!issparse(M) && issquare(M)
```

I think that we should let `\` be used even when the matrix is rectangular (`use_inv=false`). Moreover, we can add `use_pinv` as an option too.

**Comment by schillic** [comment]
> I suggest to change the default value of `use_inv`

Okay, can you do the change?

**Comment by mforets** [comment]
Yes. But on a second thought, this PR is an independent fix to `isinvertible` and we should anyways make this change 👍  

I'll change the default of `use_inv` in another PR.

File: test/unit_util.jl [hunk 1]
```diff
@@ -15,9 +15,25 @@ for _dummy_ in 1:1 # avoid global variable warnings
...
+    # dense matrix
     @test LazySets.isinvertible([2 3; 1 2])
     @test !LazySets.isinvertible([2 3; 0 0])
+    @test !LazySets.isinvertible(hcat([2 3]))
+    # sparse matrix
+    @test LazySets.isinvertible(sparse([2 3; 1 2]))
+    @test !LazySets.isinvertible(sparse([2 3; 0 0]))
+    @test !LazySets.isinvertible(sparse(hcat([2 3])))
+    # diagonal matrix
+    @test LazySets.isinvertible(Diagonal([2 0; 0 2]))
```
Comment (mforets) [inline test/unit_util.jl]: (Diagonal matrices also accept vectors: `Diagonal([2, 2])`)

**Pull request merged**